ereolen_support/ereolen_support.helper.inc

```diff
@@ -235,43 +164,30 @@ function _ereolen_material_send_to_jira($form, &$form_state, $jiraUser) {
  *   An optional data array.
  *
  * @return array
- *   An array containing base path for the request and the options related to
- *   the request.
+
  *
  * @return array
+ *   An array containing mehod, path, data and auth for the request.
+ *   the request.
  */
-  function _ereolen_support_get_request_parts($method_type, $data = [], $api = NULL) {
-  $jiraUsername = variable_get('jira_username');
-  $jiraPassword = variable_get('jira_password');
-  // Jira requires base64_encode auth.
-  $encoded_auth = base64_encode($jiraUsername . ':' . $jiraPassword);
+function _ereolen_support_get_request_parts($apiEndpoint, string $method_type, array $data = [], $api = NULL) {
```

Comment:
It seems that you always call `_ereolen_support_get_request_parts` and then call `_ereolen_support_callApi` with the result right after. I find the signature of `_ereolen_support_callApi` hard to understand, so it may make sense to combine the two functions, e.g.:

```php
function _ereolen_support_callApi($apiEndpoint, string $method_type, array $data = [], $api = NULL) {
  $requestParts = «Whatever _ereolen_support_get_request_parts() returns»

  $headers = [
    'Accept: application/json',
    'Content-Type: application/json',
  ];

  // Initialize curl.
  $curl = curl_init();

  …
}
```